test/e2e/test_e2e.py

```diff
@@ -54,9 +54,14 @@ def test_parametric_program(qc: QuantumComputer):
         results = qc.run(compiled).readout_data.get("ro")
         all_results.append(np.mean(results))
 
-    assert all_results[0] == 0.0
-    assert all_results[1] > 0.8
-    assert all_results[2] == 0.0
+    if isinstance(qc.qam, QPU):
```

Comment:
@notmgsk Does this seem reasonable, based on the circuit being run? I want to make sure I'm not hiding some real issue, as I expected an RX(0, 0) to not have any error associated with it, since it effectively becomes an I(0). But perhaps the QPU doesn't treat it as such because the compiler never made it an explicit I gate?

Reply:
The QPU is also sensitive to readout error - even a simple `MEASURE 0` program would not come out all zeroes. A threshold of 0.5 seems quite high and just asserts that we're doing better than a coin flip; something like 0.2 should handle even degraded readout while asserting that the program is likely compiling and running as expected (and would correspond with the `> 0.8` assertion for `RX(pi)`).

Reply:
Perfect. Will do.

Reply:
Done.